Contest_API.md

```diff
@@ -114,36 +114,58 @@ their expected behavior, if implemented.
   - `GET`
     Read data. This method is idempotent and does not modify any data.
     It can be used to request a whole collection or a specific element.
+
   - `POST`
     Create a new element. This can only be called on a collection
-    endpoint. No `id` attribute should be specified as it is up to the
-    server to assign one, which is returned in the location header.
+    endpoint, and the `id` attribute may not be specified as it is up
+    to the server to assign one.
+    If successful the response will contain a `Location` header
+    pointing to the newly created element.
+
   - `PUT`
-    Replaces a specific element. This method is idempotent and can only
-    be called on a specific element and replaces its contents with the
-    data provided. The payload data must be complete, i.e. no partial
-    updates are allowed. The `id` attribute cannot be changed: it does
-    not need to be specified (other than in the URL) and if specified
-    different from in the URL, a `409 Conflict` HTTP code should be
-    returned.
+    Creates or replaces a specific element. This method is idempotent, can only
+    be called on a specific element, and replaces its contents with the
+    data provided. The payload data must be complete, i.e. the `id` is
+    required and no partial updates are allowed.
+
   - `PATCH`
-    Updates/modifies a specific element. Similar to `PUT` but allows
-    partial updates by providing only that data, for example:
-    `PATCH  https://example.com/api/contests/wf14/teams/10`
-    with JSON contents
-    `{"name":"Our cool new team name"}`
-    No updates of the `id` attribute are allowed either.
+    Updates/modifies a specific element. This method is idempotent,
+    can only be called on a specific element, and replaces the given
+    attributes with the data provided. For example
+    `PATCH https://example.com/api/contests/wf14/teams/10`
+    with JSON contents `{"name":"Our cool new team name"}`.
+
   - `DELETE`
-    Delete a specific element. Idempotent, but may return a 404 status
-    code when repeated. Any provided data is ignored. Example:
-    `DELETE  https://example.com/api/contests/wf14/teams/8`
-    Note that deletes must keep [referential
-    integrity](#referential-integrity) intact.
+    Delete a specific element. Idempotent, but will return a `404` error
+    code when repeated. Any provided data is ignored and there is no response body.
+    Example: `DELETE https://example.com/api/contests/wf14/teams/8`.
+    Note that deletes must keep [referential integrity](#referential-integrity) intact.
```

Comment:
It's (a tiny bit) unclear if this note defines a requirement on the client or the server. The answer is, "the server", right? If so, this implies that the server must implement cascade or deny semantics, but it's free to choose which.

Reply:
Agreed. New commit changes it to a 'must implement fail or cascading delete', see if that works.